src/main/java/org/apache/commons/lang3/ArrayUtils.java

```diff
@@ -8672,4 +8672,53 @@ public static void shuffle(final double[] array, final Random random) {
             swap(array, i - 1, random.nextInt(i), 1);
         }
     }
+
+    /**
+     * Gets an element from the array if the array is non-null and appropriately long, otherwise returns null
+     * @param <T> the component type of the array
+     * @param array   the array holding the desired element
+     * @param index  the index of the element in the array
+     * @return The element in the array at the index, or null if it is ill-formatted
+     * @since 3.8
+     */
+    public static <T> T get(T[] array, int index){
+        return get(array, index, null);
+    }
+
+    /**
+     * Gets an element from the array if the array is non-null and appropriately long, otherwise returns the specified value
+     * @param <T> the component type of the array
+     * @param array   the array holding the desired element
+     * @param index  the index of the element in the array
+     * @param defaultReturn the object to be returned if the array is null or shorter than the index
+     * @return The element in the array at the specified index, or the given argument if it is ill-formatted
```

Comment:
ill-formatted?

Again, I don't see the point of this method.
It's a bit more flexible than the original method.
But it's still only useful for the case where you can choose a default value that does not appear in the array, and you still have to check whether the return value is the default or not.

Reply:
My thought process was something similar to this https://github.com/apache/commons-lang/blob/master/src/main/java/org/apache/commons/lang3/StringUtils.java#L7456-L7458

I would understand if Strings are understood well-enough to merit a true default value and generic arrays aren't.